pkg/kubeapiserver/authorizer/config.go

```diff
@@ -129,14 +113,6 @@ func (config AuthorizationConfig) New() (authorizer.Authorizer, authorizer.RuleR
 		default:
 			return nil, nil, fmt.Errorf("Unknown authorization mode %s specified", authorizationMode)
 		}
-		authorizerMap[authorizationMode] = true
-	}
-
-	if !authorizerMap[modes.ModeABAC] && config.PolicyFile != "" {
-		return nil, nil, errors.New("Cannot specify --authorization-policy-file without mode ABAC")
-	}
-	if !authorizerMap[modes.ModeWebhook] && config.WebhookConfigFile != "" {
-		return nil, nil, errors.New("Cannot specify --authorization-webhook-config-file without mode Webhook")
 	}
 
 	return union.New(authorizers...), union.NewRuleResolvers(ruleResolvers...), nil
```

Comment:
If you have no authorizer at this point, you must fail.  If you return without failing, you run the risk of running the cluster without any authorization.

Reply:
make sense.  But have already checked in `BuiltInAuthorizationOptions.Validate`

Reply:
> make sense. But have already checked in BuiltInAuthorizationOptions.Validate

Relying on functions to be called in order is poor design.  For preventing panics or ugly errors, calling functions in order is more acceptable.  For preventing behavioral failures, it is not.  You need to test for empty.

Reply:
ic, learned a lot

Reply:
This comment was resolved at beginning of function